src/LogicTasks/Syntax/TreeToFormula.hs

```diff
@@ -81,27 +82,29 @@ partialGrade _ sol
 
 
 
-completeGrade :: (OutputMonad m, MonadIO m) => FilePath -> SynTreeInst -> TreeFormulaAnswer -> LangM m
+completeGrade :: (OutputMonad m, MonadIO m) => FilePath -> TreeToFormulaInst -> TreeFormulaAnswer -> LangM m
 completeGrade path inst sol
-    | treeAnswer /= tree inst = refuse $ do
+    | treeAnswer /= correctTree = refuse $ do
         instruct $ do
           english "Your solution is not correct. The syntax tree for your entered formula looks like this:"
           german "Ihre Abgabe ist nicht die korrekte Lösung. Der Syntaxbaum zu Ihrer eingegebenen Formel sieht so aus:"
 
         image $=<< liftIO $ cacheTree (transferToPicture treeAnswer) path
 
         when (showSolution inst) $
-          example (show (correct inst)) $ do
+          example (show (display correctTree)) $ do
```

Comment:
I assume there was a reason (possibly Autotool convenience wise) for the correct string being persisted in the instance type, i.e., for not computing it here on the fly. I think it is best to not change this. That is, the `display` should not be computed here, instead being extracted from the precomputed instance.